primitives/xcm/src/barriers.rs

```diff
@@ -48,7 +48,8 @@ impl<T: Contains<MultiLocation>> ShouldExecute for AllowTopLevelPaidExecutionDes
 			origin, message, max_weight, _weight_credit,
 		);
 		ensure!(T::contains(origin), ProcessMessageError::Unsupported);
-		let mut iter = message.iter_mut();
+		// At most two ClearOrigin
```

Comment:
Is to check only two times if there is a`ClearOrigin` instruction within the message. By checking if there is a `ClearOrigin` instruction present, we can make the variable to be mutable and overwrite the weights properly.

Reply:
@Agusrodri a legit message will never contain any `ClearOrigin` instruction here, because the purpose of the `DescendOrigin` instruction is precisely to replace `ClearOrigin`

Reply:
> is to make the variable mutable

You shouldn't allow any ClearOrigj instruction here, and anyway it's not needed to be bale to mutate the weights

Reply:
I disagree @librelois. What is legit and what is not is not dictated by us but by use-cases that other chains can imagine. Consider the following situation.

A chain wants to have (maybe through treasury) access to a different account than the sovereign account, which for some reason holds funds (maybe it stores lost tokens and what not). In that sense, a chain might want to withdraw certain funds from the treasury-controlled account, but then might want to avoid any further action that might use the same origin. In that case, a ClearOrigin would be needed.

In my opinion this adds no harm to the current barrier. If no ClearOrigin exist the barrier will pass, and if it exists, it will pass too. I dont see why we should oppose so strongly to this

Reply:
@girazoki Yes, you're right, and the new combination with `WithComputedOrigin` allow `ClearOrigin`

Reply:
In fact, by doing `WithComputedOrigin<AllowTopLevelPaidExecutionFrom<Everything>>` you are allowing the clearOrigin, as `AllowTopLevelPaidExecutionFrom` allows it